src/components/utils/controlled_or_internal_state.ts

```diff
@@ -0,0 +1,49 @@
+import { useCallback, useEffect, useRef, useState } from 'react';
+
+function usePrevious<T>(value: T) {
+  const ref = useRef<T>(value);
+
+  useEffect(() => {
+    ref.current = value;
+  }, [value]);
+
+  return ref.current;
+}
+
+function useWarnControlled(value: unknown, name: string) {
+  const previousValue = usePrevious(value);
+  useEffect(() => {
+    if (value !== undefined && previousValue === undefined) {
+      // eslint-disable-next-line no-console
+      console.warn(
+        `SplitPane: "${name}" prop changes from being uncontrolled to being controlled. This may lead to unexpected behavior.`,
+      );
+    }
+    if (value === undefined && previousValue !== undefined) {
+      // eslint-disable-next-line no-console
+      console.warn(
+        `SplitPane: "${name}" prop changes from being controlled to being uncontrolled. This may lead to unexpected behavior.`,
+      );
+    }
+  }, [value, previousValue, name]);
+}
+
+export function useControlledOrInternalState<T>(
```

Comment:
In another project, we depend on `@radix-ui/react-use-controllable-state`. Have you considered it?

Reply:
I had a vague memory about it but did not know where to look for.
I will try to use that.